### PR TITLE
GEODE-6000: make pdx-name required

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/JdbcDistributedTest.java [hunk 1]
```diff
@@ -259,18 +259,19 @@ public void verifyDateToDate() throws Exception {
...
     });
     createRegionUsingGfsh(true, false, true);
     createJdbcConnection();
-    createMapping(REGION_NAME, DATA_SOURCE_NAME);
+    createMapping(REGION_NAME, DATA_SOURCE_NAME, TestDate.class.getName());
     final String key = "emp1";
     final java.sql.Date sqlDate = java.sql.Date.valueOf("1982-09-11");
     final Date jdkDate = new Date(sqlDate.getTime());
     server.invoke(() -> {
       PdxInstance pdxEmployee1 =
```
Comment (agingade) [inline geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/JdbcDistributedTest.java]: nice to change the variable name to testDate as its not representing employee type

File: geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java [hunk 2]
```diff
@@ -114,20 +113,33 @@ public void createExistingRegionMappingFails() {
...
   }
+
+  @Test
+  public void createMappingWithoutPdxNameFails() {
+    CommandStringBuilder csb = new CommandStringBuilder(CREATE_MAPPING);
+    csb.addOption(CREATE_MAPPING__REGION_NAME, REGION_NAME);
+    csb.addOption(CREATE_MAPPING__DATA_SOURCE_NAME, "connection");
+    // NOTE: --table is optional so it should not be in the ouput but it is. See GEODE-3468.
+    gfsh.executeAndAssertThat(csb.toString()).statusIsError()
+        .containsOutput("You should specify option (--table, --pdx-name) for this command");
```
Comment (agingade) [inline geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java]: About error message, "You should" not sure if this the pattern we use across the error messages. 
How about -  "missing x y" see command help for "create mapping"

Comment (dschneider-pivotal) [reply geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java]: For improvements to this message, make your comments on GEODE-3468.
I agree with you that this message should be improved.

File: geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/JdbcDistributedTest.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/JdbcDistributedTest.java]: done

**Pull request merged**